modules/tinymce/src/core/main/ts/table/InsertTable.ts

```diff
@@ -39,11 +37,11 @@ const isPercentage = (width: string): boolean =>
   Type.isString(width) && width.indexOf('%') !== -1;
 
 const insert = (editor: Editor, columns: number, rows: number, colHeaders: number, rowHeaders: number): HTMLTableElement => {
```

Comment:
Same here can't see why the insertTable command couldn't live in the dom core. I don't think any snooker things should be in the core.

Reply:
At the time we wanted something like insert table to be independent of whether RTC was enabled. I'm not sure if that's still true given we abandoned the tables feature for 6.0

Reply:
Having a table in the content in RTC causes dom mutation due to resize handles and table cell selection and that causes a full reflow. We don't support editing tables yet in RTC but we should avoid all dom mutation. I think this would have been avoided if you some how move the selection model and table resizing out of the core and into the dom model. Even if snooker doesn't really bloat up the core here seems to just use a small part of it it's still causing issues.

- [ ] Branch out table selection and table resize handles if rtc is enabled or move them to the dom model

Reply:
The latest 4 commits are my initial efforts to move the table resize handling, table cell selection and the `mceInsertTable` command to the model. I haven't done the "get selected cells" method aspect as that introduces more complexity so will have to be discussed further tomorrow on the best way to handle that aspect

Reply:
Looking at the code I don't think we need to use commands or events for this. Unless I missed something it seems to be that different modules within the table model needs to talk to each other. That wouldn't need to be public api since commands and events become pubic api. It could be some internal pub/sub thing or shared api object between them. Or are these events needed for the advanced tables plugin?

Reply:
Thanks for the suggestions @spocke. I have removed the command concept since you are right that they aren't strictly needed. Instead I have added a `getSelectedCells` and `clearSelectedCells` to the model API which core, table plugin and the advatble plugin can then use